include/ocpp/v201/smart_charging.hpp

```diff
@@ -47,15 +47,15 @@ std::ostream& operator<<(std::ostream& os, const ProfileValidationResultEnum val
 /// to calculate the composite schedules
 class SmartChargingHandler {
 private:
-    std::map<int32_t, std::unique_ptr<EvseInterface>>& evses;
+    std::map<int32_t, std::shared_ptr<EvseInterface>>& evses;
```

Comment:
Creating a type alias for this type here like `using EvseMap = std::map<int32_t, std::unique_ptr<EvseInterface>>` and using it might make the code more readable.

Reply:
Is there any reason, why changing from `std::unique_ptr` to `std::shared_ptr`?

Reply:
I have another change in progress that gets rid of the `map<int, unique_ptr<EvseInterface>>` thing by using a different class to manage the Evse since I think this map exposes way more than needed. So I don't see the need to change the map right now. It is a good suggestion otherwise.

As for the shared ptr, it's to make mocking easier since the test driver can access the `EvseMock` as well as inject it into the `SmartChargingHandler`. This will probably also change with the above mentioned PR.

Reply:
I had a similar issue, but usually then the unit test is not written properly (I guess it is about the lifetime right? The unit test just creates evse's and forgets about them - they should be part of the unit tests context - you might want to do assertions on them later on anyway and pass them via `std::ref` do the map).  In general I wouldn't make any compromises on the ownership of objects (shared vs. unique) in the code, just to make it easier to write unit tests.
But depending on the PR you're mentioning, this problem might be gone by then.

Reply:
For reference, the PR i was talking about is here: https://github.com/EVerest/libocpp/pull/668

I will update this one after that one is merged.

Reply:
Well as you know, I really don't like changing code just for unittests. It makes things more complex to understand and read in my opinion.